spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java

```diff
@@ -310,39 +309,6 @@ public void testMergeWithMultipleUpdatesForTargetRow() {
         sql("SELECT * FROM %s ORDER BY id ASC NULLS LAST", tableName));
   }
 
-  @Test
-  public void testMergeWithDisabledCardinalityCheck() {
```

Comment:
We introduced this table property as we expected the cardinality check to be too expensive. In particular, we thought we would need a separate job for merge-on-read. However, the current implementation is cheap and does not require any extra jobs so I am not sure we need that property...

Reply:
I'll deprecate it in a separate PR since this one is so huge.